### PR TITLE
Update the Hermiticity check of `qml.Hermitian` to use `qml.math`

### PR DESCRIPTION
**Pull request by antalszava**
Reworks the Hermiticity check in `qml.Hermitian` by using `qml.math` because calling `.conj()` on an EagerTensor from TF raises an error.

Closes https://github.com/PennyLaneAI/pennylane/issues/2892

### CODE REVIEW

File: pennylane/ops/qubit/observables.py [hunk 1]
```diff
@@ -16,13 +16,15 @@
...
-from scipy.sparse import csr_matrix
+from copy import copy
 
 import numpy as np
+from scipy.sparse import csr_matrix
 
 import pennylane as qml
 from pennylane.operation import AllWires, AnyWires, Observable
 from pennylane.wires import Wires
+
```
Comment (antalszava) [inline pennylane/ops/qubit/observables.py:27]: Linting (issues with Codefactor).

**Review by Jaybsoni — APPROVED** [review]
Looks good to me! I am surprised that tf can't natively resolve this, but autoray can!

**Comment by Jaybsoni** [comment]
Maybe add a more descriptive title to the PR?

**Comment by antalszava** [comment]
[sc-24249]

**Pull request merged**